Mu2eG4/src/SimParticleHelper.cc

```diff
@@ -16,11 +16,19 @@ namespace mu2e {
                                        const art::Event* event,
                                        const art::EDProductGetter* sim_prod_getter)
     : simStage_(simStage)
-    , particleNumberOffset_(inputs.simParticleNumberOffset())
+    , particleNumberOffset_(0u)
     , simID_(simID)
     , event_(event)
     , simProductGetter_(sim_prod_getter)
-  {}
+  {
+    // particleNumberOffset_ is set per event, based on the highest number
+    // used by SimParticles in previous simulation stages.
+    if(inputs.multiStage()) {
+      const auto sph = event->getValidHandle<SimParticleCollection>(inputs.inputSimParticles());
+      particleNumberOffset_ = sph->back().first.asUint();
```

Comment:
OK I see it is an internal variable.

Reply:
There is no "global" simParticleOffset any more, but we still need to shift particle numbers in each event w.r.t. geant track numbers to keep particle IDs unuque.